feat: support multiple versions of the pg_stat_monitor extension

jfroche opened the pull request:
Build multiple versions of the pg_stat_monitor extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

jfroche committed:
feat: support multiple versions of the pg_stat_monitor extension

Build multiple versions of the pg_stat_monitor extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

samrose committed:
feat: rebase on develop and suffix for testing

hunleyd reviewed (changes requested):
rebase to fix conflicts?

hunleyd reviewed (approved):
LGTM

jfroche committed:
feat: use default nixos extension test

And run pg_regress tests for pg_stat_monitor as part of the unified extension tests

samrose reviewed (changes requested):
testing now
